### PR TITLE
Indicator types - cidr, ipv6, ipv6_cidr - 101

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/20662

## Description
initial PR, missing:
formatting scripts
indicator fields
proper validation if regex

## Required version of Demisto
5.5

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation - not yet
- [ ] Code Review

## Technical Writers:
- [ ] [CHANGELOG](https://github.com/demisto/content/blob/ed434e4a091d728c532d7b6bf0e02dcad4edf724/Misc/reputation-ip_CHANGELOG.md)

### CODE REVIEW

**Comment by bakatzir** [comment]
depends on: https://github.com/demisto/etc/issues/20955

File: Misc/reputation-ipCIDR.json [hunk 1]
```diff
@@ -0,0 +1,140 @@
+{
```
Comment (avidan-H) [inline Misc/reputation-ipCIDR.json]: Preserve consistency - filename should be all lowercase like the other reputation files.

File: Misc/reputation-ipv6.json [hunk 2]
```diff
@@ -0,0 +1,140 @@
+{
+    "id": "ipv6",
+    "version": -1,
+    "fromVersion": "5.0.0",
```
Comment (avidan-H) [inline Misc/reputation-ipv6.json]: The `fromVersion` should be `5.5.0`, no?
```suggestion
    "fromVersion": "5.5.0",
```

File: Misc/reputation-ipv6CIDR.json [hunk 3]
```diff
@@ -0,0 +1,140 @@
+{
+    "id": "ipv6_cidr",
+    "version": -1,
+    "fromVersion": "5.0.0",
```
Comment (avidan-H) [inline Misc/reputation-ipv6CIDR.json]:
```suggestion
    "fromVersion": "5.5.0",
```

File: Misc/reputation-ipv6CIDR.json [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    ],
+    "system": true,
+    "locked": false,
+    "disabled": false,
+    "file": false,
+    "updateAfter": 0,
+    "mergeContext": false,
+    "formatScript": "",
+    "contextPath": "IP(val.Address \u0026\u0026 val.Address === obj.Address)",
+    "contextValue": "Address",
```
Comment (avidan-H) [inline Misc/reputation-ipv6CIDR.json]: This definitely isn't correct.

File: Misc/reputation-ipv6.json [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        "SplunkPySearch"
+    ],
+    "system": true,
+    "locked": false,
+    "disabled": false,
+    "file": false,
+    "updateAfter": 0,
+    "mergeContext": false,
+    "formatScript": "",
+    "contextPath": "IP(val.Address \u0026\u0026 val.Address === obj.Address)",
```
Comment (avidan-H) [inline Misc/reputation-ipv6.json]:
```suggestion
    "contextPath": "IPv6(val.Address \u0026\u0026 val.Address === obj.Address)",
```

File: Misc/reputation-ipv6.json [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    "shouldCommit": false,
+    "regex": "\\b(?:(?:[0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|(?:[0-9a-fA-F]{1,4}:){1,7}:|(?:[0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|(?:[0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|(?:[0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|(?:[0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|(?:[0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:(?:(:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))\\b",
+    "details": "IPv6",
+    "prevDetails": "IPv6",
+    "reputationScriptName": "",
+    "reputationCommand": "ip",
+    "enhancementScriptNames": [
+        "IPReputation",
+        "SplunkPySearch"
+    ],
```
Comment (avidan-H) [inline Misc/reputation-ipv6.json]: should remove

File: Misc/reputation-ipCIDR.json [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    "shouldCommit": false,
+    "regex": "\\b(?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])(?:\\[\\.\\]|\\.)){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])(\\/([0-9]|[1-2][0-9]|3[0-2]))\\b",
+    "details": "IP CIDR",
+    "prevDetails": "IP CIDR",
+    "reputationScriptName": "",
+    "reputationCommand": "ip",
+    "enhancementScriptNames": [
+        "IPReputation",
+        "SplunkPySearch"
+    ],
```
Comment (avidan-H) [inline Misc/reputation-ipCIDR.json]: should remove

File: Misc/reputation-ipCIDR.json [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+    "system": true,
+    "locked": false,
+    "disabled": false,
+    "file": false,
+    "updateAfter": 0,
+    "mergeContext": false,
+    "formatScript": "",
+    "contextPath": "IP(val.Address \u0026\u0026 val.Address === obj.Address)",
+    "contextValue": "Address",
```
Comment (avidan-H) [inline Misc/reputation-ipCIDR.json]: This isn't correct.

**Comment by bakatzir** [comment]
@avidan-H `contextPath` will be decided in the future by  @cloughrm, along with the indicator fields. thus for now, it's just the same as `IP`(IPv4)

**Comment by cloughrm** [comment]
Names should be:
IPv6
CIDR
IPv6 CIDR

File: Misc/reputation-cidr.json [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+                    }
+                ]
+            }
+        },
+        "reportedby": {
+            "simple": "",
+            "complex": {
+                "root": "CIDR",
+                "filters": [],
+                "accessor": "Reported By",
```
Comment (avidan-H) [inline Misc/reputation-cidr.json]:
```suggestion
                "accessor": "ReportedBy",
```

File: Misc/reputation-cidr.json [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+                    }
+                ]
+            }
+        },
+        "threattypes": {
+            "simple": "",
+            "complex": {
+                "root": "CIDR",
+                "filters": [],
+                "accessor": "Threat Types",
```
Comment (avidan-H) [inline Misc/reputation-cidr.json]:
```suggestion
                "accessor": "ThreatTypes",
```

File: Misc/reputation-cidr.json [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+                    }
+                ]
+            }
+        },
+        "trafficlightprotocoltlp": {
+            "simple": "",
+            "complex": {
+                "root": "CIDR",
+                "filters": [],
+                "accessor": "Traffic Light Protocol (TLP)",
```
Comment (avidan-H) [inline Misc/reputation-cidr.json]:
```suggestion
                "accessor": "TrafficLightProtocol",
```

File: Misc/reputation-ip.json [hunk 12]
```diff
@@ -133,6 +133,104 @@
...
+                    }
+                ]
+            }
+        },
+        "reportedby": {
+            "simple": "",
+            "complex": {
+                "root": "IP",
+                "filters": [],
+                "accessor": "Reported By",
```
Comment (avidan-H) [inline Misc/reputation-ip.json]:
```suggestion
                "accessor": "ReportedBy",
```

File: Misc/reputation-ip.json [hunk 13]
```diff
@@ -133,6 +133,104 @@
...
+                    }
+                ]
+            }
+        },
+        "threattypes": {
+            "simple": "",
+            "complex": {
+                "root": "IP",
+                "filters": [],
+                "accessor": "Threat Types",
```
Comment (avidan-H) [inline Misc/reputation-ip.json]:
```suggestion
                "accessor": "ThreatTypes",
```

File: Misc/reputation-ip.json [hunk 14]
```diff
@@ -133,6 +133,104 @@
...
+                    }
+                ]
+            }
+        },
+        "trafficlightprotocoltlp": {
+            "simple": "",
+            "complex": {
+                "root": "IP",
+                "filters": [],
+                "accessor": "Traffic Light Protocol (TLP)",
```
Comment (avidan-H) [inline Misc/reputation-ip.json]:
```suggestion
                "accessor": "TrafficLightProtocol",
```

File: Misc/reputation-ipv6.json [hunk 15]
```diff
@@ -0,0 +1,235 @@
...
+                    }
+                ]
+            }
+        },
+        "reportedby": {
+            "simple": "",
+            "complex": {
+                "root": "IPv6",
+                "filters": [],
+                "accessor": "Reported By",
```
Comment (avidan-H) [inline Misc/reputation-ipv6.json]:
```suggestion
                "accessor": "ReportedBy",
```

File: Misc/reputation-ipv6.json [hunk 16]
```diff
@@ -0,0 +1,235 @@
...
+                    }
+                ]
+            }
+        },
+        "threattypes": {
+            "simple": "",
+            "complex": {
+                "root": "IPv6",
+                "filters": [],
+                "accessor": "Threat Types",
```
Comment (avidan-H) [inline Misc/reputation-ipv6.json]:
```suggestion
                "accessor": "ThreatTypes",
```

File: Misc/reputation-ipv6.json [hunk 17]
```diff
@@ -0,0 +1,235 @@
...
+                    }
+                ]
+            }
+        },
+        "trafficlightprotocoltlp": {
+            "simple": "",
+            "complex": {
+                "root": "IPv6",
+                "filters": [],
+                "accessor": "Traffic Light Protocol (TLP)",
```
Comment (avidan-H) [inline Misc/reputation-ipv6.json]:
```suggestion
                "accessor": "TrafficLightProtocol",
```

File: Misc/reputation-ipv6_cidr.json [hunk 18]
```diff
@@ -0,0 +1,212 @@
...
+                    }
+                ]
+            }
+        },
+        "reportedby": {
+            "simple": "",
+            "complex": {
+                "root": "IPv6CIDR",
+                "filters": [],
+                "accessor": "Reported By",
```
Comment (avidan-H) [inline Misc/reputation-ipv6_cidr.json]:
```suggestion
                "accessor": "ReportedBy",
```

File: Misc/reputation-ipv6_cidr.json [hunk 19]
```diff
@@ -0,0 +1,212 @@
...
+                    }
+                ]
+            }
+        },
+        "threattypes": {
+            "simple": "",
+            "complex": {
+                "root": "IPv6CIDR",
+                "filters": [],
+                "accessor": "Threat Types",
```
Comment (avidan-H) [inline Misc/reputation-ipv6_cidr.json]:
```suggestion
                "accessor": "ThreatTypes",
```

File: Misc/reputation-ipv6_cidr.json [hunk 20]
```diff
@@ -0,0 +1,212 @@
...
+                    }
+                ]
+            }
+        },
+        "trafficlightprotocoltlp": {
+            "simple": "",
+            "complex": {
+                "root": "IPv6CIDR",
+                "filters": [],
+                "accessor": "Traffic Light Protocol (TLP)",
```
Comment (avidan-H) [inline Misc/reputation-ipv6_cidr.json]:
```suggestion
                "accessor": "TrafficLightProtocol",
```

**Pull request merged**